### DISCUSSION TITLE
Calibration can cause the addition of trends that did not really happen

### DISCUSSION THREAD

**Discussion by thedewi** [Navid200/xDrip#133] [General]
Until recently, when I added a calibration, xDrip+ would update the last half hour to reconstruct a realistic gradient. Recipients of the data (Nightscout, Garmin watchface) would show a discontinuity, which is fair enough.

However, my xDrip+ instance is currently combining old and new values which can create trends that did not occur.

# Example

I am experiencing a firmly downward trend towards a possible hypo. This is reflected in the raw values: 

![Screenshot_20211202-145018~2](https://user-images.githubusercontent.com/4726656/144361035-9ccb48b4-f91e-452c-8231-1c7b1cdffc0e.png)

I have applied a calibration that reveals the downward trend is less severe, but xDrip is graphing those 6 points as though they abruptly changed direction and entered an upward trend, when that didn't happen: the raw data was consistently downward.

![Screenshot_20211202-144259](https://user-images.githubusercontent.com/4726656/144361414-69e807cc-87f3-48b1-8780-fd1dcad221ec.png)

Does anyone understand why it's doing this?

**Comment by thedewi** [comment]
To clarify my expectation:

I preferred the old behaviour which would reinterpret old raw values against the latest calibrations gradient.

It would also be fine to leave points where they first landed.

What's happening now appears to be some kind of "fade" between the outdated calibration gradient and the new, so I can no longer see the real trend.

**Comment by thedewi** [comment]
Version is 49817a5-2021.10.04.

**Comment by Navid200** [comment]
Please check:
Settings -> xDrip+ Display Settings -> Graph Settings -> Rewrite History

If it is disabled, you can enable it to get the behavior you desire.

**Reply by thedewi** [reply]
It is disabled, and I will try with it on. It's strange that it it already rewriting history with that turned off.

**Reply by Navid200** [reply]
One relatively recent change was that the default was changed from enabled to disabled.
So, if you did a re-install, that may explain it.

**Reply by thedewi** [reply]
Happy to report that enabling "Rewrite History" makes it work the way that makes sense to me.
